runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -59,6 +60,61 @@ class MemoryManager final {
     bool contains(void *ptr) { return _impl.contains(ptr); }
 };
 
+class SharedLibraryManager final {
+  private:
+    void *_handler{NULL};
+
+  public:
+    SharedLibraryManager() = delete;
+    SharedLibraryManager(std::string filename)
+    {
+        // RTLD_DEEPBIND is incompatible with sanitizers.
+        // If you have compiled this file with sanitizers and you reach this line
+        // you will get an error.
+        // Please re-compile without sanitizers.
+        _handler = dlopen(filename.c_str(), RTLD_LAZY | RTLD_DEEPBIND);
+        RT_FAIL_IF(!_handler, dlerror());
```

Comment:
Please don't forget to use the macros defined in `Exceptions.hpp` :)

Reply:
These are macros defined in Exceptions.hpp.

Reply:
Ah! I made this comment before.. it looks good! :+1: